data_structures/heap/skew_heap.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+from typing import Generic, Iterable, List, Optional, TypeVar
+
+__all__ = ["SkewHeap"]
+
+T = TypeVar("T")
+
+
+class SkewNode(Generic[T]):
+    """One node of the skew heap. Contains the value and references to two children."""
+
+    def __init__(self, value: T) -> None:
+        self._value: T = value
+        self.left: Optional[SkewNode[T]] = None
+        self.right: Optional[SkewNode[T]] = None
+
+    @property
+    def value(self) -> T:
+        """Return the value of the node."""
+        return self._value
+
+    @staticmethod
+    def merge(
+        root1: Optional[SkewNode[T]], root2: Optional[SkewNode[T]]
+    ) -> Optional[SkewNode[T]]:
+        """Merge 2 nodes together."""
+        if not root1:
+            return root2
+
+        if not root2:
+            return root1
+
+        if root1.value > root2.value:
+            root1, root2 = root2, root1
+
+        result = root1
+        temp = root1.right
+        result.right = root1.left
+        result.left = SkewNode.merge(temp, root2)
+
+        return result
+
+
+class SkewHeap(Generic[T]):
+    """
+    A data structure that allows inserting a new value and to pop the smallest
+    values. Both operations take O(logN) time where N is the size of the structure.
+    - Wiki: https://en.wikipedia.org/wiki/Skew_heap
+    - Visualisation: https://www.cs.usfca.edu/~galles/visualization/SkewHeap.html
+
+    >>> SkewHeap.from_list([2, 3, 1, 5, 1, 7]).to_sorted_list()
+    [1, 1, 2, 3, 5, 7]
+
+    >>> sh = SkewHeap()
+    >>> sh.insert(1)
```

Comment:
Please test `.pop()` __before__ `insert()`ing any data to ensure that it behaves like `[].pop()`.

Insert data out of order and then test to ensure that `.to_sorted_list()` returns data in order.

Please add tests for `.clear()`, `.from_list()`, and `bool()`.